parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordReader.java

```diff
@@ -201,6 +210,57 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
     return internalReader.nextKeyValue();
   }
 
+  /**
+   * Reads the next batch of rows. The caller needs to check the returned batch size with {@link parquet.vector.RowBatch#size()}.
+   * @param previous a row batch object to be reused by the reader if possible
+   * @return the row batch that was read
+   * @throws java.io.IOException
+   */
+  public RowBatch nextBatch(RowBatch previous) throws IOException {
+    MessageType requestedSchema = readContext.getRequestedSchema();
+    List<ColumnDescriptor> columns = requestedSchema.getColumns();
+    int nColumns = columns.size();
+    ColumnVector[] columnVectors;
+    if (previous == null || previous.getColumns() == null) {
+      previous = new RowBatch();
```

Comment:
if `previous` is not null and `previous.getColumns()` is null then you enter this if block and create an unnecessary new `RowBatch` instance, instead you should be using `previous` in this case.